### PR TITLE
`azurerm_key_vault_certificate`: expose arm resource id 

### PR DESCRIPTION
**Pull request by wuxu92**
resolves: #21261 

this just works as keyvault secret:

https://github.com/hashicorp/terraform-provider-azurerm/blob/a872bf3bb49bdcedb15186e50f9b57da382af1b7/internal/services/keyvault/key_vault_secret_resource.go#L372-L373

```
--- PASS: TestAccKeyVaultCertificate_basicImportPFX (440.87s)
PASS
--- PASS: TestAccDataSourceKeyVaultCertificate_basic (437.98s)
PASS
```

### CODE REVIEW

File: internal/services/keyvault/key_vault_certificate_data_source.go [hunk 1]
```diff
@@ -39,6 +39,16 @@ func dataSourceKeyVaultCertificate() *pluginsdk.Resource {
...
 
+			"resource_id": {
+				Computed: true,
+				Type:     pluginsdk.TypeString,
+			},
+
+			"resource_versionless_id": {
+				Computed: true,
+				Type:     pluginsdk.TypeString,
+			},
```
Comment (stephybun) [inline internal/services/keyvault/key_vault_certificate_data_source.go:50]: We should be more explicit in what kind of ID we're referring to so can we rename these to

```suggestion
			"resource_manager_id": {
				Computed: true,
				Type:     pluginsdk.TypeString,
			},

			"resource_manager_versionless_id": {
				Computed: true,
				Type:     pluginsdk.TypeString,
			},
```

I'm aware we have fields called `resource_id` and `resource_versionless_id` in some other key vault resources - these will need to be renamed to `resource_manager_id` and `resource_manager_versionless_id` for consistency in 4.0

File: internal/services/keyvault/key_vault_certificate_resource.go [hunk 2]
```diff
@@ -376,6 +376,16 @@ func resourceKeyVaultCertificate() *pluginsdk.Resource {
...
 
+			"resource_id": {
+				Computed: true,
+				Type:     pluginsdk.TypeString,
+			},
+
+			"resource_versionless_id": {
+				Computed: true,
+				Type:     pluginsdk.TypeString,
+			},
```
Comment (stephybun) [inline internal/services/keyvault/key_vault_certificate_resource.go:387]: Same here 
```suggestion
			"resource_manager_id": {
				Computed: true,
				Type:     pluginsdk.TypeString,
			},

			"resource_manager_versionless_id": {
				Computed: true,
				Type:     pluginsdk.TypeString,
			},
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @wuxu92 - some minor remarks and the docs also need to be updated but otherwise this looks good.

File: internal/services/keyvault/key_vault_certificate_data_source.go (hunk 1 above)
Comment (wuxu92) [reply internal/services/keyvault/key_vault_certificate_data_source.go:50]: thanks for your information. i have renamed these fileds.

**Review by stephybun — COMMENTED** [review]
Thanks for renaming those @wuxu92. Docs still need to be updated before we can merge this!

**Review by stephybun — APPROVED** [review]
Thanks @wuxu92 LGTM 💯

**Pull request merged**